compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUCheckLinalgVectorized.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "mlir/Dialect/Linalg/Utils/Utils.h"
+#include "mlir/Pass/Pass.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+struct LLVMCPUCheckLinalgVectorizedPass
+    : LLVMCPUCheckLinalgVectorizedBase<LLVMCPUCheckLinalgVectorizedPass> {
+  void runOnOperation() override;
+};
+}  // namespace
+
+void LLVMCPUCheckLinalgVectorizedPass::runOnOperation() {
+  auto funcOp = getOperation();
+  auto walkResult = funcOp.walk([&](linalg::LinalgOp op) -> WalkResult {
+    return WalkResult::interrupt();
+  });
+  if (walkResult.wasInterrupted()) {
+    funcOp.emitWarning("one or more operations were found not vectorized");
```

Comment:
Nit: using LLVM terminology, this would be something like "emitting vectorization remarks". Perhaps, using "emit" and "remarks" in the class and flag would make it clearer.